OpenDreamClient/Rendering/DreamViewOverlay.cs

```diff
@@ -1,57 +1,58 @@
 ﻿using Robust.Client.Graphics;
 using Robust.Client.Player;
 using Robust.Shared.Enums;
-using Robust.Shared.GameObjects;
-using Robust.Shared.IoC;
-using Robust.Shared.Maths;
-using System.Collections.Generic;
 using JetBrains.Annotations;
+using Robust.Shared.Map;
 
 namespace OpenDreamClient.Rendering {
-    class DreamViewOverlay : Overlay {
+    sealed class DreamViewOverlay : Overlay {
         private IPlayerManager _playerManager = IoCManager.Resolve<IPlayerManager>();
         private IEntitySystemManager _entitySystem = IoCManager.Resolve<IEntitySystemManager>();
         private IEntityManager _entityManager = IoCManager.Resolve<IEntityManager>();
+        private IMapManager _mapManager = IoCManager.Resolve<IMapManager>();
         private RenderOrderComparer _renderOrderComparer = new RenderOrderComparer();
         [CanBeNull] private EntityLookupSystem _lookupSystem;
+        [CanBeNull] private SharedTransformSystem _transformSystem;
 
         public override OverlaySpace Space => OverlaySpace.WorldSpace;
 
         protected override void Draw(in OverlayDrawArgs args) {
-            DrawingHandleWorld handle = args.WorldHandle;
-            EntityUid? eye = _playerManager.LocalPlayer.Session.AttachedEntity;
+            EntityUid? eye = _playerManager.LocalPlayer?.Session.AttachedEntity;
             if (eye == null) return;
 
-            DrawMap(handle, eye.Value);
+            DrawingHandleWorld handle = args.WorldHandle;
+            DrawMap(args, eye.Value);
             DrawScreenObjects(handle, eye.Value, args.WorldAABB);
         }
 
-        private void DrawMap(DrawingHandleWorld handle, EntityUid eye) {
-
+        private void DrawMap(OverlayDrawArgs args, EntityUid eye) {
 
+            _transformSystem ??= _entitySystem.GetEntitySystem<SharedTransformSystem>();
             _lookupSystem ??= _entitySystem.GetEntitySystem<EntityLookupSystem>();
+            var spriteQuery = _entityManager.GetEntityQuery<DMISpriteComponent>();
+            var xformQuery = _entityManager.GetEntityQuery<TransformComponent>();
 
             var entities = _lookupSystem.GetEntitiesInRange(eye, 15);
```

Comment:
If the eye is attached to the player entity then you could just use the OverlayDrawArgs Mapid + WorldAABB instead?

Reply:
That does appear to be a better way to do it, wasn't aware of it

Reply:
Either MapID or WorldAABB were just added in the most recent engine update I think so probably not available sooner.